src/backend/src/modules/auth/auth.controller.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  Body,
+  Controller,
+  Get,
+  HttpCode,
+  HttpException,
+  HttpStatus,
+  Post,
+  Redirect,
+  Req,
+} from '@nestjs/common';
+import { ApiTags, ApiOkResponse } from '@nestjs/swagger';
+
+import { Unprotected } from 'nest-keycloak-connect';
+import { AuthService } from './auth.service';
+import { KeycloakToken } from './keycloack-token.model';
+
+@Controller('auth')
+@ApiTags('Auth')
+export class AuthController {
+  constructor(private authService: AuthService) {}
+
+  @Get('keycloakLogin')
+  @Redirect('', HttpStatus.MOVED_PERMANENTLY)
+  @Unprotected()
+  @ApiOkResponse({
+    description: 'redirect to keycloak sso server for user login',
+  })
+  login() {
+    return this.authService.getUrlLogin();
+  }
+
+  @Post('callback')
+  @Unprotected()
+  @ApiOkResponse({ description: 'Get access token payload with credentials' })
+  async getAccessToken(@Body('code') code: string) {
+    if (code === '') return;
```

Comment:
A better looking solution to this can be this (coving empty strings, null responses, etc)
```suggestion
    if (!code) return;
```